components/insight/SRC/org/openmicroscopy/shoola/util/roi/io/ROIReader.java

```diff
@@ -342,58 +342,9 @@ private void setImage(Roi[] rois)
                     }
                 }
             } else if (r instanceof ShapeRoi) {
-                Roi[] subRois = ((ShapeRoi) r).getRois();
-                Roi shapeij;
-                for (int j = 0; j < subRois.length; j++) {
-                    shapeij = subRois[j];
-
-                    // Set ImagePlus reference in subROIs for the check in L216 to work
-                    ImagePlus imp = r.getImage();
-                    shapeij.setImage(imp);
-                    // Transfer correct ROI positions (according to IJ) from superROI
-                    int pos = r.getPosition();
-                    int c = r.getCPosition();
-                    int z = r.getZPosition();
-                    int t = r.getTPosition();
-                    if (imp.getNChannels() == 1 && imp.getNSlices() == 1) {
-                        shapeij.setPosition(pos);
-                    } else if (imp.getNChannels() == 1 &&
-                            imp.getNFrames() == 1) {
-                        shapeij.setPosition(pos);
-                    } else if (imp.getNSlices() == 1 &&
-                            imp.getNFrames() == 1) {
-                        shapeij.setPosition(pos);
-                    } else if (imp.isHyperStack()) {
-                        shapeij.setPosition(c, z, t);
-                    }
-                    type = shapeij.getTypeAsString();
```

Comment:
Ahh apologies, I did not see the read() method properly. And I was assuming that the composite ROI's are similar to Overlays, and thus replicated your testing workflow without doing "Add to ROI-Manager" : thus the lack of ROI's in the resultant image.

Corrected the workflow now and I can see the resultant ROI's. 

But one small issue, there seems to be a ghost (invisible) ROI as well. I drew six roi's (composite) in my workflow but I keep getting a Seventh ghost ROI (which is invisible in the clients and the count seems to be 1 more than the expected number in the Omero side). This happens only for Composite ROI's and only on the omero Side (Please Check ROI Count for the Image in the omero clients), and not for the regular ROI's or Overlay workflows. Do you see this in your testing workflow as well? If its an issue at our end alone, I can dig deeper and troubleshoot the issue in better detail.

Reopening the image on ImageJ gives me just 6 ROI's, so I just wanted to check with you on this regard, before digging deeper into the issue.